weblate/static/editor/full.js

```diff
@@ -549,23 +607,65 @@
       if (typeof el.origin !== "undefined") {
         service.append(" (");
         var origin;
+        var deleteUrl = false;
         if (typeof el.origin_detail !== "undefined") {
           origin = $("<abbr/>").text(el.origin).attr("title", el.origin_detail);
         } else if (typeof el.origin_url !== "undefined") {
           origin = $("<a/>").text(el.origin).attr("href", el.origin_url);
         } else {
           origin = el.origin;
         }
+        if (el.delete_url) {
+          deleteUrl = true;
+        }
         service.append(origin);
         service.append(")");
       }
-      return service;
+      return {
+        service: service,
+        deleteUrl: deleteUrl,
+      };
+    }
+
+    renderDeleteUrls(text) {
+      var translations = this.state.translations;
+      var modalBody = $("<label>").text("");
+
+      translations.forEach((translation) => {
+        if (translation.service === "Weblate Translation Memory") {
+          if (
+            text === translation.text &&
+            typeof translation.delete_url !== "undefined"
+          ) {
```

Comment:
Is the condition for service name needed? Maybe it's better to filter by the delete_url only?

```suggestion
          if (
            text === translation.text &&
            typeof translation.delete_url !== "undefined"
          ) {
```

Reply:
okay, I added the condition to be more explicit. Now, I am thinking only the `"Weblate Translation Memory"` will have delete_url so the later check would be enough. I will remove the first one. 👍